repos/system_upgrade/el8toel9/actors/nischeck/libraries/nischeck.py

```diff
@@ -0,0 +1,64 @@
+from leapp import reporting
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM, NISConfig
+
+report_summary = (
+    'NIS components (ypserv, ypbind, yp-tools) will no longer be available in RHEL-9.'
+    ' Technology behind those packages is based an outdated design patterns no longer'
+    ' considered secure. There is no direct alternative with fully compatible features.'
+)
+
+report_hint = (
+    'The alternatives are LDAP and for some use cases Kerberos. Migrating to IPA'
+    ' is also a solution.'
+)
+
+report_link_url = 'https://access.redhat.com/solutions/5991271'
+
+
+def report_nis():
+    """
+    Create the report if any of NIS packages (RH signed)
+    is installed and configured.
+
+    Should notify user about present NIS compnent package
+    installation, warn them about discontinuation, and
+    redirect them to online documentation for possible
+    alternatives.
+    """
+
+    installed_packages = []
+    configured_rpms = []
+
+    # Get necessary models created by another actors
+    model_nis_conf = api.consume(NISConfig)
+
+    # Get and filter information from models
+    for model in model_nis_conf:
+        configured_rpms = model.nis_not_default_conf
```

Comment:
This silently relies on receiving exactly one `NISConfig` message. IMHO, if receiving exactly one message is expected, it should be explicit from the code. 
```suggestion
    nis_confs = api.consume(NISConfig)
    nis_conf = next(nis_confs, None)
    if not nis_conf:
        # Not receiving any NISConfig is highly unlikely, and it is OK to terminate Leapp by raising StopActorExecution informing about no messages received
    if next(nis_confs, None):
        # Log a warning about receiving multiple NIS configs
   configured_rpms = nis_conf.nis_not_default_conf
```

Reply:
It's also fine and might be more readable for someone. Changed.